services/mapper.go

```diff
@@ -197,13 +199,28 @@ func FeeOps(tx *evmClient.LoadedTransaction) []*RosettaTypes.Operation {
 	return ops
 }
 
+func getCurrency(params []interface{}) *RosettaTypes.Currency {
+	var nativeCurrency *RosettaTypes.Currency
+	if len(params) > 0 {
+		if currencyParam, ok := params[0].(*RosettaTypes.Currency); ok {
+			nativeCurrency = currencyParam
+		}
+	}
+	if nativeCurrency == nil {
+		nativeCurrency = sdkTypes.Currency
+	}
+	return nativeCurrency
+}
+
 // TraceOps returns all *RosettaTypes.Operation for a given
 // array of flattened traces.
 // nolint:gocognit
 func TraceOps(
 	calls []*evmClient.FlatCall,
 	startIndex int,
+	args ...interface{},
```

Comment:
this function argument change can potentially break all upstream Rosetta (e.g. Base, Arbitrum, BSC). regression tests for different chains are needed for this change